src/backend/base/langflow/base/models/aiml_constants.py

```diff
@@ -1,252 +1,46 @@
-    "teknium/OpenHermes-2-Mistral-7B",
-    "teknium/OpenHermes-2p5-Mistral-7B",
-    "togethercomputer/CodeLlama-13b-Instruct",
-    "togethercomputer/CodeLlama-13b-Python",
-    "togethercomputer/CodeLlama-34b",
-    "togethercomputer/CodeLlama-34b-Instruct",
-    "togethercomputer/CodeLlama-34b-Python",
-    "togethercomputer/CodeLlama-7b-Instruct",
-    "togethercomputer/CodeLlama-7b-Python",
-    "togethercomputer/Koala-13B",
-    "togethercomputer/Koala-7B",
-    "togethercomputer/LLaMA-2-7B-32K",
-    "togethercomputer/Llama-2-7B-32K-Instruct",
-    "togethercomputer/Llama-3-8b-chat-hf-int4",
-    "togethercomputer/Llama-3-8b-chat-hf-int8",
-    "togethercomputer/SOLAR-10.7B-Instruct-v1.0-int4",
-    "togethercomputer/StripedHyena-Hessian-7B",
-    "togethercomputer/StripedHyena-Nous-7B",
-    "togethercomputer/alpaca-7b",
-    "togethercomputer/evo-1-131k-base",
-    "togethercomputer/evo-1-8k-base",
-    "togethercomputer/guanaco-13b",
-    "togethercomputer/guanaco-33b",
-    "togethercomputer/guanaco-65b",
-    "togethercomputer/guanaco-7b",
-    "togethercomputer/llama-2-13b",
-    "togethercomputer/llama-2-13b-chat",
-    "togethercomputer/llama-2-70b",
-    "togethercomputer/llama-2-70b-chat",
-    "togethercomputer/llama-2-7b",
-    "togethercomputer/llama-2-7b-chat",
-    "togethercomputer/m2-bert-80M-2k-retrieval",
-    "togethercomputer/m2-bert-80M-32k-retrieval",
-    "togethercomputer/m2-bert-80M-8k-retrieval",
-    "upstage/SOLAR-10.7B-Instruct-v1.0",
-    "voyage-2",
-    "voyage-code-2",
-    "voyage-finance-2",
-    "voyage-large-2",
-    "voyage-large-2-instruct",
-    "voyage-law-2",
-    "voyage-multilingual-2",
-    "wavymulder/Analog-Diffusion",
-    "zero-one-ai/Yi-34B",
-    "zero-one-ai/Yi-34B-Chat",
-    "zero-one-ai/Yi-6B",
-]
+from typing import Any
 
-AIML_EMBEDDING_MODELS = [
-    "text-embedding-3-large",
-    "text-embedding-3-small",
-    "text-embedding-ada-002",
-    "text-multilingual-embedding-002",
-    "textembedding-gecko-multilingual@001",
-    "textembedding-gecko@001",
-    "textembedding-gecko@003",
-]
+import httpx
+
+
+class AimlModels:
+    def __init__(self):
+        self.chat_models = []
+        self.image_models = []
+        self.embedding_models = []
+        self.stt_models = []
+        self.tts_models = []
+        self.language_models = []
+
+    def get_aiml_models(self) -> list[dict[str, Any]]:
+        try:
+            with httpx.Client() as client:
+                response = client.get("https://api.aimlapi.com/models")
+                response.raise_for_status()
+                models = response.json().get("data", [])
+                self.separate_models_by_type(models)
+        except httpx.RequestError as e:
+            self.get_exception_message(e)
+        except httpx.HTTPStatusError as e:
+            self.get_exception_message(e)
+        except ValueError as e:
+            self.get_exception_message(e)
+
+    def separate_models_by_type(self, models):
```

Comment:
Gabriel raises a good point - at the moment we don't need many of these models. IIRC, aiml exposes per-model-type endpoints, so you can retrieve just the chat models or embedding models? I would recommend separating the `GET`s so you can make specific model retrievals per-type as needed.

Reply:
It's a good point, agreed - but for now we do not have per-model-type endpoints. Hence, why such sorting is needed